pkg/testutils/testutils.go

```diff
@@ -0,0 +1,227 @@
+package testutils
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	configv1 "github.com/openshift/api/config/v1"
+	cloudingressv1alpha1 "github.com/openshift/cloud-ingress-operator/pkg/apis/cloudingress/v1alpha1"
+
+	awsclient "github.com/openshift/cloud-ingress-operator/pkg/awsclient/mock"
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const masterMachineLabel string = "machine.openshift.io/cluster-api-machine-role"
+
+const DefaultRegionName string = "us-east-1"
+const DefaultAzName string = "us-east-1a"
+const DefaultAPIEndpoint string = "https://api.unit.test:6443"
+const DefaultClusterDomain string = "unit.test"
+
+// ClusterTokenId represents the part of identifiers which is varied by the installer, eg
+// clustername-clustertokenid, as in load balancer names: foo-12345-us-east-1a
+const ClusterTokenId string = "12345"
+
+type Mocks struct {
+	FakeKubeClient client.Client
+	MockCtrl       *gomock.Controller
+	MockAws        *awsclient.MockClient
+	Scheme         *runtime.Scheme
+}
+
+// NewMockTest sets up for a new mock test, pass in some localObjs to seed the fake Kubernetes environment
+func NewTestMock(t *testing.T, localObjs []runtime.Object) *Mocks {
+	mockctrl := gomock.NewController(t)
+	s := scheme.Scheme
+	if err := configv1.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add configv1 scheme: (%v)", err)
+	}
+	if err := machineapi.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add machine scheme: (%v)", err)
+	}
+	if err := cloudingressv1alpha1.SchemeBuilder.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add cloudingressv1alpha1 scheme: (%v)", err)
+	}
```

Comment:
This method looks like a straight lift-and-shift from [1] -- except that this block used to be first. It looks like that's not a significant change logically, right?

No change suggested, just trying to understand why it moved.

[1] https://github.com/openshift/cloud-ingress-operator/pull/50/files#diff-e422e4e0f311f6f2a6e15f011a8e1b18L40-L62

Reply:
The order changed for a rather mundane reason: The `testutils` package came about while writing for `utils` tests, and they didn't need the cloud ingress operator schemata. Once the `testutils` package was ready to be backported into the apischeme controller, I added it after the other scheme adds. No real importance.